backend/metadata_writer/src/metadata_writer.py

```diff
@@ -15,16 +15,21 @@
 import json
 import hashlib
 import os
-import sys
 import re
+import glob
 import kubernetes
 import yaml
 from time import sleep
+from lru import LRU
 
 from metadata_helpers import *
 
 
 namespace_to_watch = os.environ.get('NAMESPACE_TO_WATCH', 'default')
+pod_name_to_execution_id_size = os.environ.get('POD_NAME_TO_EXECUTION_ID_SIZE', 5000)
+workflow_name_to_context_id_size = os.environ.get('WORKFLOW_NAME_TO_CONTEXT_ID_SIZE', 5000)
+pods_with_written_metadata_size = os.environ.get('PODS_WITH_WRITTEN_METADATA_SIZE', 5000)
+debug_files_size = os.environ.get('DEBUG_FILES_SIZE', 5000)
```

Comment:
I've exposed these settings as env vars so folks if needs be can tweak these settings if they want to cache more/less aggressively via `kustomize` patches.